pulsar-connect/core/src/main/java/org/apache/pulsar/connect/core/PushSource.java

```diff
@@ -48,5 +48,5 @@
      * to pass messages whenever there is data to be pushed to Pulsar.
      * @param consumer
      */
-    void setConsumer(Function<T, CompletableFuture<Void>> consumer);
-}
\ No newline at end of file
+    void setConsumer(Function<T, CompletableFuture<Void>> consumer) throws Exception;
```

Comment:
why do we need this change?

Reply:
if the user wants to throw checked exception but not wanting to wrap it around a try catch and throw runtimeException.  Its kinda unnecessary